### DISCUSSION TITLE
New (1.4.0) web UI

### DISCUSSION THREAD

**Discussion by puuu** [kanidm/kanidm#3187] [General]
Thank you for creating this software, I have been testing it since version 1.3.3.

Since 1.4.0 the new web UI is integrated, in general I like it, but I would like to suggest some small adjustments to make it more user friendly.

When logging in, you are asked for your username and there is a switch to remember your username.
But what is the point of this switch if it is only for the browser session (the expiration time of the cookie is the session)?

There is a discussion in #1542 about usernameless webauthn and short/straight login.
Why in v1.4.0 do we now get an additional "Use passkey" button after entering the username, but only passkeys are registered as credentials?
If we really need this step, please make this single button autofocus. It is somehow annoying to be forced to use a mouse or two keys (Tab and Enter) for this step.

A similar problem occurs when adding a password. First the focus is on "Enter New Password", then Tab takes us to "Repeat Password". This is fine, but Enter does not do "Submit".

I also miss the "Add new/another device" option from 1.3.3, which gives us a QR.
The [documentation](/kanidm/kanidm/blob/master/book/src/developers/designs/device-authentication.rst) describes how to use this feature.
At first I did not understand it from the UI, but after reading the documentation I am now missing this feature.

**Comment by Firstyear** [comment]
> Thank you for creating this software, I have been testing it since version 1.3.3.
> 
> Since 1.4.0 the new web UI is integrated, in general I like it, but I would like to suggest some small adjustments to make it more user friendly.
> 
> When logging in, you are asked for your username and there is a switch to remember your username. But what is the point of this switch if it is only for the browser session (the expiration time of the cookie is the session)?

Hmmm I thought we made this a permanent cookie, but I'll fix this asap. 

> 
> There is a discussion in #1542 about usernameless webauthn and short/straight login. Why in v1.4.0 do we now get an additional "Use passkey" button after entering the username, but only passkeys are registered as credentials? If we really need this step, please make this single button autofocus. It is somehow annoying to be forced to use a mouse or two keys (Tab and Enter) for this step.

It's because of a bug in some browsers, where you have to interact to actually trigger webauthn. That's why we need this extra click. We are working to improve this flow though. 

> 
> A similar problem occurs when adding a password. First the focus is on "Enter New Password", then Tab takes us to "Repeat Password". This is fine, but Enter does not do "Submit".
> 
> I also miss the "Add new/another device" option from 1.3.3, which gives us a QR. The [documentation](/kanidm/kanidm/blob/master/book/src/developers/designs/device-authentication.rst) describes how to use this feature. At first I did not understand it from the UI, but after reading the documentation I am now missing this feature.

We should re-add this. Sorry it was dropped.

**Reply by carlitoplatanito** [reply]
> It's because of a bug in some browsers, where you have to interact to actually trigger webauthn. That's why we need this extra click. We are working to improve this flow though.

That's actually a security "*feature*", hah. You can set the button to autofocus though, the user would still have to hit `enter` (or click on it), but it would avoid the initial `tab`.

> **Disclaimer:** Just discovered Kanidm yesterday and it's 🔥 so far! Just happened to open this discussion hoping it was about adding a webUI for admin user/group management 🥲

**Reply by Firstyear** [reply]
All the issues in your previous comment should have been resolved now in 1.4.5 :)

**Comment by Firstyear** [comment]
@puuu Sorry for the delay, we are re-adding this add another device feature now. #3301
